services/src/apidoc.rs

```diff
@@ -0,0 +1,133 @@
+    VectorResultDescriptor,
+};
+use utoipa::openapi::security::{HttpAuthScheme, HttpBuilder, SecurityScheme};
+use utoipa::{Modify, OpenApi};
+
+#[derive(OpenApi)]
+#[openapi(
+    handlers(
+        crate::server::show_version_handler,
+        handlers::session::anonymous_handler,
+        handlers::workflows::register_workflow_handler,
+        handlers::workflows::load_workflow_handler,
+        handlers::workflows::get_workflow_metadata_handler,
+        handlers::workflows::get_workflow_provenance_handler,
+        handlers::workflows::dataset_from_workflow_handler,
+    ),
+    components(
+        SimpleSession,
+
+        DataId,
+        DataProviderId,
+        DatasetId,
+        ExternalDataId,
+        IdResponse<WorkflowId>,
+        LayerId,
+        ProjectId,
+        SessionId,
+        UploadId,
+        WorkflowId,
+
+        TimeInstance,
+        TimeInterval,
+
+        Coordinate2D,
+        BoundingBox2D,
+        SpatialPartition2D,
+        SpatialResolution,
+        SpatialReference,
+        SpatialReferenceOption,
+        SpatialReferenceAuthority,
+        Measurement,
+        ContinuousMeasurement,
+        ClassificationMeasurement,
+        STRectangle,
+
+        ProvenanceOutput,
+        Provenance,
+
+        VectorDataType,
+        RasterDataType,
+
+        VersionInfo,
+
+        Workflow,
+        TypedOperator,
+        TypedResultDescriptor,
+        PlotResultDescriptor,
+        RasterResultDescriptor,
+        VectorResultDescriptor,
+        RasterDatasetFromWorkflow,
+        RasterDatasetFromWorkflowResult,
+        RasterQueryRectangle,
+        VectorQueryRectangle,
+        PlotQueryRectangle,
+    ),
+    modifiers(&SecurityAddon, &ApiDocInfo),
+    external_docs(url = "https://docs.geoengine.io", description = "Geo Engine Docs")
+)]
+pub struct ApiDoc;
+
+struct SecurityAddon;
+
+impl Modify for SecurityAddon {
+    fn modify(&self, openapi: &mut utoipa::openapi::OpenApi) {
+        let components = openapi.components.as_mut().unwrap();
+        components.add_security_scheme(
+            "session_token",
+            SecurityScheme::Http(
+                HttpBuilder::new()
+                    .scheme(HttpAuthScheme::Bearer)
+                    .bearer_format("UUID")
+                    .description(Some("A valid session token can be obtained via the /anonymous or /login (pro only) endpoints. Alternatively, it can be defined as a fixed value in the Settings.toml file."))
+                    .build(),
+            ),
+        );
+    }
+}
+
+struct ApiDocInfo;
+
+impl Modify for ApiDocInfo {
+    fn modify(&self, openapi: &mut utoipa::openapi::OpenApi) {
+        openapi.info.title = "Geo Engine API".to_string();
+
+        openapi.info.contact = Some(
+            utoipa::openapi::ContactBuilder::new()
+                .name(Some("Geo Engine Developers"))
+                .email(Some("dev@geoengine.de"))
```

Comment:
Should/can we link to the repo here?

Reply:
it's possible to add a url. But I am not sure if we want to link the repo, or our website. Especially in deployments we might want to point to the operator of the instance 🤷